src/EnergyPlus/UnitarySystem.hh

```diff
@@ -0,0 +1,684 @@
+                                      int &CompOn,                               // compressor on/off control
+                                      Optional<Real64 const> OAUCoilOutTemp = _, // the coil inlet temperature of OutdoorAirUnit
+                                      Optional_bool HXUnitOn = _                 // Flag to control HX for HXAssisted Cooling Coil
+        );
+
+        void updateUnitarySystemControl(int const AirLoopNum,    // number of the current air loop being simulated
+                                        int const OutNode,       // coil outlet node number
+                                        int const ControlNode,   // control node number
+                                        Real64 &OnOffAirFlowRatio,
+                                        bool const FirstHVACIteration,
+                                        Optional<Real64 const> OAUCoilOutletTemp = _, // "ONLY" for zoneHVAC:OutdoorAirUnit
+                                        Optional<Real64> ZoneLoad = _,
+                                        Optional<Real64 const> MaxOutletTemp = _ // limits heating coil outlet temp [C]
+        );
+
+        void initLoadBasedControl(int const AirLoopNum,    // number of the current air loop being simulated
+                                  bool const FirstHVACIteration,
+                                  Real64 &OnOffAirFlowRatio,
+                                  Real64 &ZoneLoad
+        );
+
+        void sizeUnitarySystem(bool const FirstHVACIteration,
+                               int const AirLoopNum // does this need to be optional?
+        );
+
+        void setOnOffMassFlowRate(Real64 &OnOffAirFlowRatio, // ratio of coil on to coil off air flow rate
+                                  Real64 const PartLoadRatio // coil part-load ratio
+        );
+
+        void setAverageAirFlow(Real64 const PartLoadRatio, // unit part load ratio
+                               Real64 &OnOffAirFlowRatio   // ratio of compressor ON airflow to AVERAGE airflow over timestep
+        );
+
+        void calcUnitarySystemToLoad(int const AirLoopNum,              // index to air loop
+                                     bool const FirstHVACIteration,     // True when first HVAC iteration
+                                     Real64 const CoolPLR,              // operating cooling part-load ratio []
+                                     Real64 const HeatPLR,              // operating cooling part-load ratio []
+                                     Real64 &OnOffAirFlowRatio,         // ratio of heating PLR to cooling PLR (is this correct?)
+                                     Real64 &SensOutput,                // sensible capacity (W)
+                                     Real64 &LatOutput,                 // latent capacity (W)
+                                     Optional_bool HXUnitOn = _,        // Flag to control HX for HXAssisted Cooling Coil
```

Comment:
I'd personally prefer that we start moving new code away from these ObjexxFCL `Optional`s.

Reply:
That was the last thing I did (3 or 4 days ago) and am now trying to correct the code to run with all the arguments.